### PR TITLE
Fix environment switch glitches

### LINKED ISSUE #1021: Environment maps not loading correctly after switching environments
## Hardware
HTC Vive Focus (could not reproduce on Oculus Go)
FxR v1.1.3a release armeabi-v7a

## Steps to Reproduce & Current Behavior
Bring up settings->environments. Switch between environments one or more times. Observe that one or more sides of cube map doesn't load. Reset environment. Try again. One or more sides of cubemaps, possibly different from first attempt doesn't load.

## Expected Behavior
Cubemaps should load all sides.

## Error Logs and Stack Traces
<details open>

```
019-03-15 23:22:08.990 9336-9514/? I/VRCORE: OnReceiveEvent: event.eventType=202
2019-03-15 23:22:08.990 9336-9514/? I/VRCORE: OnReceiveCoreEvent: event.eventType = 202, event.inputId = 9, event.deviceType = 1, devType = 2, event.interaction_mode = 0, event.gaze_trigger = 0
2019-03-15 23:22:08.990 9336-9514/? I/VRCORE: OnReceiveEvent: event.eventType=200
2019-03-15 23:22:08.990 9336-9514/? I/VRCORE: OnReceiveCoreEvent: event.eventType = 200, event.inputId = 9, event.deviceType = 1, devType = 2, event.interaction_mode = 0, event.gaze_trigger = 0
2019-03-15 23:22:09.170 9336-9514/? I/VRCORE: OnReceiveEvent: event.eventType=201
2019-03-15 23:22:09.170 9336-9514/? I/VRCORE: OnReceiveCoreEvent: event.eventType = 201, event.inputId = 9, event.deviceType = 1, devType = 2, event.interaction_mode = 0, event.gaze_trigger = 0
2019-03-15 23:22:09.172 9336-9514/? I/VRCORE: OnReceiveEvent: event.eventType=203
2019-03-15 23:22:09.172 9336-9514/? I/VRCORE: OnReceiveCoreEvent: event.eventType = 203, event.inputId = 9, event.deviceType = 1, devType = 2, event.interaction_mode = 0, event.gaze_trigger = 0
2019-03-15 23:22:09.190 819-1277/? W/AudFlgTr: track uid 10071 buf is zero
2019-03-15 23:22:09.194 9336-9512/? I/VRB: Setting environment: cubemap/meadow
2019-03-15 23:22:09.219 9336-9605/? I/VRB: Cached data: 188 size: 524288
2019-03-15 23:22:09.222 9336-9605/? I/VRB: Cached data: 189 size: 524288
2019-03-15 23:22:09.222 9336-9512/? I/VRB: Loaded cached data: 188 size: 524288
2019-03-15 23:22:09.223 9336-9512/? I/VRB: Loaded cached data: 189 size: 524288
2019-03-15 23:22:09.225 9336-9605/? I/VRB: Cached data: 190 size: 524288
2019-03-15 23:22:09.227 9336-9605/? I/VRB: Cached data: 191 size: 524288
2019-03-15 23:22:09.230 9336-9605/? I/VRB: Cached data: 192 size: 524288
2019-03-15 23:22:09.233 9336-9605/? I/VRB: Cached data: 193 size: 524288
2019-03-15 23:22:09.249 9336-9605/? I/VRB: TIMER Accumulative time loading assets: 1.047177 sec
2019-03-15 23:22:09.492 842-10210/? W/QCamera3HWI: num_output_buffers = 1 input_buffer = 0x0 frame_number = 3180

```

</details>

device-2019-03-15-232845
device-2019-03-15-230839

### PR DESCRIPTION
**Pull request by MortimerGoro**
Fixes #1021

The problem is that TextureCubemap internal initilization is triggered from two threads at the same time, the loader thread, and the main render thread when it binds the cubemap texture for the current frame. This happens because we are using a geometry already added to the scenegraph after a environment change.

### CODE REVIEW

File: app/src/main/cpp/Skybox.cpp [hunk 1]
```diff
@@ -144,7 +142,15 @@ struct Skybox::State {
       return group;
     };
 
-    loader->RunLoadTask(transform, task);
+    vrb::GeometryPtr oldGeometry = geometry;
+    LoadFinishedCallback loadedCallback = [=](GroupPtr&) {
+      geometry->GetRenderState()->SetTintColor(tintColor);
```
Comment (bluemarvin) [inline app/src/main/cpp/Skybox.cpp]: Should `geometry` be checked for `nullptr`?

**Comment by bluemarvin** [comment]
>  environments appear to load significantly faster on the Pico G2 - nice work.

The Pico G2 CPU is significantly faster than the Go.

**Comment by cvan** [comment]
> >  environments appear to load significantly faster on the Pico G2 - nice work.
> 
> The Pico G2 CPU is significantly faster than the Go.

my phrasing was ambiguous: before this PR, the environments changed faster on the Oculus Go than on the Pico G2. in this branch, the environments load at perceptibly the same speed. good improvements across the board :+1:

**Comment by MortimerGoro** [comment]
> before this PR, the environments changed faster on the Oculus Go than on the Pico G2. in this branch, the environments load at perceptibly the same speed. good improvements across the board

It makes sense, the environment was loaded twice due to the initialization race condition

**Pull request merged**